### ISSUE TITLE
Add support for fasmg in clang/lib/Driver/ToolChains

### ISSUE THREAD

**Issue by adriweb** [jacobly0/llvm-z80#8]
Right now, only the ZDS assembler is supported, but that's been made obsolete with fasmg, so it'd be nice to be able to use that.

Would also make things easier for TI-Planet's Project Builder when I decide to finally update the toolchain...

**Comment by cesss** [comment]
Adding support for any Z80 assembler that doesn't require Windows (nor being run through Wine) would be really welcome.

**Comment by adriweb** [comment]
Yeah [fasmg](https://travis-ci.org/jacobly0/fasmg-ez80/) (which is for ez80 at least) runs natively on Windows, macOS, and Linux.

[spasmg-ng](https://github.com/alberthdev/spasm-ng) can also be built natively on those platforms.
